tests/test_butler.py

```diff
@@ -101,6 +102,18 @@ def mock_s3(cls):
 TESTDIR = os.path.abspath(os.path.dirname(__file__))
 
 
+def clean_environment() -> None:
+    """Remove external environment variables that affect the tests."""
+    for k in (
+        "DAF_BUTLER_REPOSITORY_INDEX",
+        "S3_ENDPOINT_URL",
+        "AWS_ACCESS_KEY_ID",
+        "AWS_SECRET_ACCESS_KEY",
+        "AWS_SHARED_CREDENTIALS_FILE",
+    ):
+        os.environ.pop(k, None)
```

Comment:
Would it make sense to make this more reusable and move to `python/lsst/daf/butler/tests/utils.py`?

Reply:
I'm half inclined to remove the S3 code from here completely but we need one remote datastore for testing and we don't use the webdav one by default. There is `lsst.resources.s3utils` that tries to have something but it's not quite what we need. I probably should be trying to make that work for the S3 use case at USDF (where pytest doesn't work without this change). The DAF_BUTLER_REPOSITORY_INDEX clean up is really only needed for this one test of `testConstructor` and a general utility for the S3 tests in butler is only relevant for this one file.